eng/common/scripts/git-branch-push.ps1

```diff
@@ -60,8 +60,14 @@ if ($LASTEXITCODE -ne 0)
     exit $LASTEXITCODE
 }
 
-Write-Host "git checkout -b $PRBranchName"
-git checkout -b $PRBranchName
+# Check if the branch is default branch in remote.
+$defaultBranch = (git remote show $RemoteName | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
```

Comment:
Why is this check needed here? I assume we can checkout the branch still even if it is the default branch.

Reply:
`git checkout -b master` will fail if $PRBranchName is default branch.

Reply:
I don't think that is just for the default branch I suspect that is true for any case where you are already on the PRBranchName so maybe the check should be more geared towards that instead of only the default branch. Or we can also just handle any errors that come when trying to checkout the branch in question.

Reply:
Lets remove this change completely unless there is a know issue and if there is I'd be interested in a link to the failures it causes.